R/fixed_regex_linter.R

```diff
@@ -61,46 +61,28 @@ fixed_regex_linter <- function() {
     xml <- source_expression$xml_parsed_content
 
     patterns <- xml2::xml_find_all(xml, xpath)
+    pattern_strings <- get_r_string(patterns)
+    is_static <- is_not_regex(pattern_strings)
+
+    fixed_equivalent <- get_fixed_string(pattern_strings[is_static])
+    msg <- paste(
+      "This regular expression is static, i.e., its matches can be expressed as a fixed substring expression, which",
+      "is faster to compute.",
+      "Here, you can use", encodeString(fixed_equivalent, quote = '"', justify = "none"), "as a fixed substring.",
+      "For static regular expression patterns, set fixed = TRUE or use stringr::fixed()."
```

Comment:
let's customize this too while we're at it -- either recommend fixed=TRUE or stringr::fixed() depending on the regex flavor

Reply:
With this in hand I think we can dramatically simplify the lint message:

This regex pattern pattern is equivalent to X with fixed=TRUE.

maybe adding "it is more efficient to use fixed=TRUE when possible and typically easier to read."

wdyt @bersbersbers -- would the message be clear enough if the fixed equivalent is provided?

I guess the purely no special character case would also warrant special handling then...

Reply:
Honestly I like the general advice first, special information second.
I'm fine with dynamic choice of `fixed = TRUE` vs. `stringr::fixed()` but wouldn't go further.

Reply:
I'm fine with either - I think outputting `fixed_equivalent` helps a lot already.